.github/workflows/release.yaml

```diff
@@ -6,7 +6,26 @@ on:
       - "*"
 
 jobs:
+  pre_release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get plugin version
+        id: get_plugin_version
+        run: echo "PLUGIN_VERSION=$(cat plugin.yaml | grep "version" | cut -d '"' -f 2)" >> $GITHUB_OUTPUT
+      - name: Get tag name
+        id: get_tag_name
+        run: echo "TAG_NAME=$(echo ${{ github.ref_name }} | cut -d 'v' -f2)" >> $GITHUB_OUTPUT
+    outputs:
+      plugin_version: ${{ steps.get_plugin_version.outputs.PLUGIN_VERSION }}
+      tag_name: ${{ steps.get_tag_name.outputs.TAG_NAME }}
+
   release:
+    needs: pre_release
+    if: needs.pre_release.outputs.tag_name == needs.pre_release.outputs.plugin_version
```

Comment:
If it does not match - will it fail explicitly, so we know the problem exists? Or will it silently skip it?
The intention is to fail explicitly.

Reply:
It will silently skip it, i will change it so it will fail